lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
So, with this payload, JSON API is valid

```
{
  "data": {
    "id": "1",
    "type": "my-type",
    "attributes": {
      "name": "my name"
    },
    "relationships": {
      "posts": {
      }
    }
  }
}
```

So this looks like it may be an issue with ember?